File: experiments/matrix.java
// Implement matrix multiply in Java
// Be sure to add some mechanism to time the experiment result.

class Matrix {
  static int N = 1024;
  static double[][] A = new double[N][N];
  static double[][] B = new double[N][N];
  static double[][] C = new double[N][N];

  public static void main(String[] args) {

    // Start our timers
    double start, end;
    start = System.currentTimeMillis();

    // Initialize array
    for (int i = 0; i < N; ++i) {
      for (int j = 0; j < N; ++j) {
        A[i][j] = Math.random();
        B[i][j] = Math.random();
        C[i][j] = Math.random();
      }
    }

    // Perform multiplication
    for (int i = 0; i < N; ++i) {
      for (int j = 0; j < N; ++j) {
        for (int k = 0; k < N; ++k) {
          C[i][j] += A[i][k] * B[k][j];
        }
      }
    }

    end = System.currentTimeMillis();
    System.out.println(String.valueOf((end - start) / 1000));
  }
}

File: experiments/matrix.py
import sys, random
from time import *

start = time()

n = 1024


A = [[random.random() for row in range(n)]for col in range(n)]
B = [[random.random() for row in range(n)]for col in range(n)]
C = [[0 for row in range(n)]for col in range(n)]

# Record time of our experiment
for i in range(n):
    for j in range(n):
        for k in range(n):
            C[i][j] += A[i][k] * B[k][j]

# Stop time of our experiment
end = time()
print("%0.5f" % (end-start))
